src/bytes.ts

```diff
@@ -459,3 +459,27 @@ export function createDataView(bytes: Uint8Array): DataView {
 
   return new DataView(bytes.buffer, bytes.byteOffset, bytes.byteLength);
 }
+
+/**
+ * Compares two Uint8Arrays in a timing-safe manner.
+ *
+ * @param a - The first Uint8Array to compare.
+ * @param b - The second Uint8Array to compare.
+ * @returns Whether the Uint8Arrays are equal.
+ */
+export function areUint8ArraysEqual(a: Uint8Array, b: Uint8Array): boolean {
+  const viewA = createDataView(a);
+  const viewB = createDataView(b);
+
+  const maxLength = Math.max(a.byteLength, b.byteLength);
+
+  let diff = Math.abs(a.byteLength - b.byteLength);
+
+  for (let i = 0; i < maxLength; i++) {
+    const aByte = i < a.byteLength ? viewA.getUint8(i) : 0;
+    const bByte = i < b.byteLength ? viewB.getUint8(i) : 0;
+    diff += Math.abs(aByte - bByte);
+  }
+
+  return diff === 0;
+}
```

Comment:
Did you compare this with the function we have in `snaps-utils` for number arrays? It should work for byte arrays as well AFAIK and is a lot simpler 🤔

https://github.com/MetaMask/snaps/blob/69f2a5a2849698d58130964eb59449404b35c544/packages/snaps-utils/src/array.ts

Reply:
Yeah, to use this I would have to do some extra conversion on my byte arrays to fit `unknown[]`. Also my use case is comparing mnemonics, from my understanding it's generally better to use a timing-safe operation for comparing mnemonics/secrets.

Reply:
Sorry, didn't realise that before. Not sure if timing is actually an issue here since you're not doing actual cryptography, maybe check with the security team. Either way, I feel like we could create a more generic version of your function, e.g., support `ArrayLike`. I don't think using `DataView` is necessary.

Reply:
That function still ends up not being timing-safe because of the early exit. I'll check with the security team, but I still think this function could be useful regardless.

Reply:
Bitwise operations are disallowed via the eslint config, I used that implementation before